uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/utils/FuseServerManipulator.java

```diff
@@ -358,9 +358,11 @@ public static void addModule(String server, String project) {
 	 */
 	public static void setImmeadiatelyPublishing(String name, boolean value) {
 
-		new ServersView2().getServer(name).addAndRemoveModules();
-		new DefaultShell("Add and Remove...");
-		FuseModifyModulesPage page = new FuseModifyModulesPage();
+		ServersView2 view = new ServersView2();
+		view.open();
+		view.getServer(name).addAndRemoveModules();
+		DefaultShell shell = new DefaultShell("Add and Remove...");
+		FuseModifyModulesPage page = new FuseModifyModulesPage(shell);
```

Comment:
method addAndRemoveModules() returns ModifyModulesDialog which can be passed to the constructor FuseModifyModulesPage